### PR TITLE
rewrite

### PR DESCRIPTION
**Pull request by markedwardmurray**
@jmburges

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,108 @@
...
+
+  * `allLocationNames` that takes no arguments and returns an `NSArray`,
+  * `locationWithName:` that takes one `NSString` argument called `name` and returns a `FISLocation` object, and
+  * `locationsNearLatitude:longitude:margin:` that takes three `CGFloat` arguments called `latitude`, `longitude`, and `margin`, and returns an `NSArray`.
+
+Define these methods to return `nil`.
+
+On the `FISLocation` class, declare five methods:
+
+  * `init` — the default initializer that will be overridden,
```
Comment (misterfifths) [inline README.md:46]: I'm generally not a fan of re-declaring inherited methods in the header.

File: README.md [hunk 2]
```diff
@@ -1,25 +1,108 @@
...
+  * `initWithName:latitude:longitude:` — a designated initializer,
+  * `stringByTruncatingNameToLength:` which takes one `NSUInteger` argument called `length` and returns an `NSString`,
+  * `validateLocationData` which takes no arguments and returns a `BOOL`, and
```
Comment (misterfifths) [inline README.md]: `isValid`? It's obvious that it deals with location stuff -- it's on the location class.

File: README.md [hunk 3]
```diff
@@ -1,25 +1,108 @@
...
+
+Define the initializers to assign to `self` a call of `[super init]` and return `self`.
+
+Run the tests with `⌘` `U` to see that they fail.
+
+### III. Build the `FISTrivium` Class
+
+1. Define `FISTrivium`'s designated initializer.
+
+2. Override `FISTrivium`'s default initializer to call the designated initializer. Have it pass in an empty string (`@""`) for the `content` argument, and `0` (zero) for the `likes` argument.
```
Comment (misterfifths) [inline README.md]: I kind of like the idea of being a little less direct about telling them what to do. Instead describe the goal:
"Override `FISTrivium`'s default initializer to call the designated initializer. Default instances of `FISTrivium` should have an empty string for their `content` and no `likes`."

File: README.md [hunk 4]
```diff
@@ -1,25 +1,108 @@
...
+Run the test in `FISTriviumSpec.m` to check the `FISTrivium` class.
+
+### IV. Build the `FISLocation` Class
+
+#### Write the Initializers
+
+1. Define `FISLocation`'s designated initializer.  
+**Hint:** *Don't forget to set the* `trivia` *property to the initialization of an empty mutable array.*
+
+2. Override `FISLocation`'s default initializer to call the designated initializer. Have it pass in an empty string (`@""`) for the `name` argument, and `0.0` for both the `latitude` and `longitude` arguments.
```
Comment (misterfifths) [inline README.md]: Same note as above -- too direct

File: README.md [hunk 5]
```diff
@@ -1,25 +1,108 @@
...
 
-  8. Add an `NSMutableArray` property to `FISLocation` to hold trivia items
-  9. Write a method called `topTrivia` that returns the trivia item with the most likes. Where should this method go?
-  10. Create an initializer for `FISTrivia` that takes an argument for all of the properties and sets them appropriately
+1. Define the `allLocationNames` method to return an array of all of the `name` string properties of the `FISLocation` objects held in the `locations` array property.
 
+2. Define the `locationWithName:` method to return a `FISLocation` object whose `name` string property matches the string submitted to the `name` argument. If there isn't a match, this method should return `nil`.  
+**Hint:** *You can accomplish with either a* `for in` *loop &* `if` *statement, or by using* `NSPredicate`.
 
+3.	Define the `locationsNearLatitude:longitude:margin:` method to return an array of all of the locations whose coordinates are near enough to the submitted coordinates to fall within the submitted positive or negative margin submitted in the `margin` argument.
```
Comment (misterfifths) [inline README.md]: ".. all the locations whose coordinates are within ±`margin` degrees from the given location"

File: README.md [hunk 6]
```diff
@@ -1,25 +1,108 @@
...
+On the `FISAppDelegate` class add a public property:
+
+  * an `NSMutableArray` called `locations`.
+
+### II. Declare Methods and Define Initial Implementations
+
+On the `FISAppDelegate` class, declare three methods:
+
+  * `allLocationNames` that takes no arguments and returns an `NSArray`,
+  * `locationWithName:` that takes one `NSString` argument called `name` and returns a `FISLocation` object, and
```
Comment (misterfifths) [inline README.md]: `locationNamed:`

File: locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m [hunk 7]
```diff
@@ -0,0 +1,24 @@
+//  FISTrivium.m
+
+
+#import "FISTrivium.h"
+
+@implementation FISTrivium
+
+- (instancetype)init {
+    self = [self initWithContent:@""
+                           likes:0   ];
```
Comment (misterfifths) [inline locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m:10]: I'm really not a fan of this sort of indentation.. painful to maintain and not any more readable in my mind

File: locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    return self;
+}
+
+- (instancetype)initWithContent:(NSString *)content
+                          likes:(NSUInteger)likes {
+    self = [super init];
+    if (self) {
+        _content = content;
+        _likes=likes;
```
Comment (misterfifths) [inline locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m]: spaces for consistency's sake

File: README.md (hunk 1 above)
Comment (markedwardmurray) [inline README.md:46]: @jmburges what's your opinion on this? I remember learning this as good practice but I can't explicitly say from where. Could've just been an opinion I found on the internet.

File: locationTrivia-ObjectsTests/FISLocationSpec.m [hunk 9]
```diff
@@ -16,99 +10,119 @@
...
+            empireState.name = nil;
+            expect([empireState validateLocationData]).to.beFalsy();
         });
     });
 
-    describe(@"Shorten Location Name", ^{
-        it(@"Should have a method called shortenLocationNameToLength:", ^{
-            expect(location).to.respondTo(@selector(shortenedNameToLength:));
+    describe(@"stringByTruncatingNameToLength:", ^{
+        it(@"should return a 6-letter substring of the location's name when length equals 6", ^{
```
Comment (misterfifths) [inline locationTrivia-ObjectsTests/FISLocationSpec.m:106]: Worth adding a check for truncating to a length > the length of the string? I think `-substringToIndex:` crashes in that case.

File: README.md (hunk 2 above)
Comment (markedwardmurray) [inline README.md]: sounds like a getter. `validateData` ?

**Comment by misterfifths** [comment]
first review done!

File: README.md (hunk 3 above)
Comment (markedwardmurray) [inline README.md]: :+1:

File: README.md (hunk 2 above)
Comment (misterfifths) [inline README.md]: it is a getter, for all intents and purposes. If I were making this class, I'd make it a readonly property.

File: locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m (hunk 8 above)
Comment (markedwardmurray) [inline locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m]: oops

File: locationTrivia-ObjectsTests/FISLocationSpec.m (hunk 9 above)
Comment (markedwardmurray) [inline locationTrivia-ObjectsTests/FISLocationSpec.m:106]: great point, i didn't test for that.

File: README.md (hunk 2 above)
Comment (markedwardmurray) [inline README.md]: i had that thought as well. where should the private setter get called? in the initializer? the unit omits discussion of overwriting setters.

Comment (misterfifths) [inline README.md]: I wouldn't worry about _actually_ making it a property at this point. Just declare that method and let it be.

Comment (markedwardmurray) [inline README.md]: good enough for me.

File: locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m (hunk 7 above)
Comment (markedwardmurray) [inline locationTrivia-Objects/locationTrivia-Objects/FISTrivium.m:10]: IMO it helps with readability on Github that doesn't employ line wrap.

**Comment by markedwardmurray** [comment]
resolves #1 #2 #3 #4 #5

**Pull request merged**